benchmark/webstreams/readable-read.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+const common = require('../common.js');
+const { ReadableStream } = require('node:stream/web');
+
+const bench = common.createBenchmark(main, {
+  n: [1e5],
+  type: ['normal', 'byob'],
+});
+
+async function main({ n, type }) {
+  switch (type) {
+    case 'normal': {
+      const rs = new ReadableStream({
+        pull: function(controller) {
+          controller.enqueue('a');
+        },
+      });
+      const reader = rs.getReader();
+      let x = null;
+      bench.start();
+      for (let i = 0; i < n; i++) {
+        const { value } = await reader.read();
+        x = value;
+      }
+      bench.end(n);
+      console.assert(x);
+      break;
+    }
+    case 'byob': {
+      const encode = new TextEncoder();
+      const rs = new ReadableStream({
+        type: 'bytes',
+        pull: function(controller) {
+          controller.enqueue(encode.encode('a'));
```

Comment:
Same case as `Uint8Array`, maybe we can cache this `encode.encode` just to be sure any performance improvement on `TextEncoder` will not affect this benchmark.

Reply:
Same problem as uint8array below encode also returns fresh buffers everytime we call it and i believe those get detached and cant be used again to enqueue so the following code
```mjs
const encode = new TextEncoder();
const encoded = encode.encode('a');
const rs = new ReadableStream({
  type: 'bytes',
  pull: function (controller) {
    controller.enqueue(encoded);
  },
});
const reader = rs.getReader({ mode: 'byob' });
let x = null;
for (let i = 0; i < 100; i++) {
  const { value } = await reader.read(new Uint8Array(1));
  x = value;
}
console.assert(x);
```
gives the same kinda error:
```console
node:internal/errors:496
    ErrorCaptureStackTrace(err);
    ^

TypeError [ERR_INVALID_STATE]: Invalid state: chunk ArrayBuffer is zero-length or detached
    at new NodeError (node:internal/errors:405:5)
    at ReadableByteStreamController.enqueue (node:internal/webstreams/readablestream:1149:13)
    at Object.pull (file:///Users/debadreechatterjee/Documents/personal/node/test.mjs:6:16)
    at ensureIsPromise (node:internal/webstreams/util:185:19)
    at readableByteStreamControllerCallPullIfNeeded (node:internal/webstreams/readablestream:3043:5)
    at node:internal/webstreams/readablestream:3048:9 {
  code: 'ERR_INVALID_STATE'
}

Node.js v20.6.1
```